src/ts/sign.ts

```diff
@@ -139,6 +147,33 @@ function ecdsaSignOrder(
   }
 }
 
+function ecdsaSignOrderCancellation(
```

Comment:
I like this idea, it also applies for combining the repeating code in `hashOrderCancellation` and `hashOrder`.

Reply:
Great suggestion here. I have made all the suggestions but please notice that the function signature for `signTypedData` wasn't as generic as we might have hoped. This is mostly because of the non-uniform way we treat the data being signed in each case. We wound up having to pass `eip712Value` and `ethSignValue` as follows:

```js
function ecdsaSignTypedData(
  scheme: EcdsaSigningScheme,
  owner: Signer,
  domain: TypedDataDomain,
  types: Record<string, TypedDataField[]>,
  eip712Value: NormalizedOrder | OrderCancellation,
  ethSignValue: string,
): Promise<string> {
```

I will take a closer look at this soon and see if we can find something better here.